src/crs_linter/cli.py

```diff
@@ -114,199 +84,254 @@ def generate_version_string(directory):
+        error = True
+
+    for d in diff:
+        d = d.strip("\n")
+        r = re.match(r"^@@ -(\d+),(\d+) \+\d+,\d+ @@$", d)
+        if r:
+            line1, line2 = [int(i) for i in r.groups()]
+            e = {
+                "indent": 2,
+                "file": filename,
+                "title": "Indentation error",
+                "line": line1,
+                "endLine": line1 + line2,
+                "message": "an indentation error has found",
+            }
+            errmsgf(e)
+        logger.error(d.strip("\n"))
+
+    return error
+
+
+def read_files(filenames):
+    global logger
+
+    parsed = {}
+    # filenames must be in order to correctly detect unused variables
+    filenames = sorted(filenames)
+
+    for f in filenames:
         try:
-            with open(f, 'r') as inputfile:
-                data = inputfile.read()
+            with open(f, "r") as file:
+                data = file.read()
                 # modify the content of the file, if it is the "crs-setup.conf.example"
                 if f.startswith("crs-setup.conf.example"):
                     data = remove_comments(data)
         except:
-            errmsg("Can't open file: %s" % f)
+            logger.error(f"Can't open file: {f}")
             sys.exit(1)
 
         ### check file syntax
-        msg("Config file: %s" % (f))
+        logger.info(f"Config file: {f}")
         try:
             mparser = msc_pyparser.MSCParser()
             mparser.parser.parse(data)
-            msg(" Parsing ok.")
-            parsed_structs[f] = mparser.configlines
+            logger.debug(f"Config file: {f} - Parsing OK")
+            parsed[f] = mparser.configlines
         except Exception as e:
             err = e.args[1]
-            if err['cause'] == "lexer":
+            if err["cause"] == "lexer":
                 cause = "Lexer"
             else:
                 cause = "Parser"
-            errmsg("Can't parse config file: %s" % (f))
-            errmsgf({
-                'indent': 2,
-                'file': f,
-                'title': "%s error" % (cause),
-                'line': err['line'],
-                'endLine': err['line'],
-                'message': "can't parse file"})
-            retval = 1
+            logger.error(
+                f"Can't parse config file: {f}",
+                title=f"{cause} error",
+                file=f,
+                line=err["line"],
+                end_line=err["line"],
+            )
             continue
 
-    msg("Checking parsed rules...")
-    crsver = ""
-    for f in parsed_structs.keys():
+    return parsed
+
+
+def parse_args(argv):
+    parser = argparse.ArgumentParser(
+        prog="crs-linter", description="CRS Rules Check tool"
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        dest="output",
+        default="native",
+        help="Output format",
+        choices=["native", "github"],
+        required=False,
+    )
+    parser.add_argument(
```

Comment:
This argument isn't documented in [README.md](https://github.com/coreruleset/crs-linter/blob/3f10697fa1b5536f27d202093a38ad30269ab5d1/README.md) (even it's mandatory).

Reply:
Good catch. Now being a tool that lives _outside_ the coreruleset directory, it needs it to properly catch the git labels.

It might not be needed if we are passing the proper version 🤔